app/lib/widgets/farm_item.dart

```diff
@@ -48,24 +142,97 @@ class _FarmItemWidgetState extends State<FarmItemWidget> {
       children: [
         ListTile(
           title: TextField(
-              readOnly: true,
+              focusNode: walletFocus,
+              autofocus: edit,
+              readOnly: !edit,
               style: Theme.of(context).textTheme.bodyMedium!.copyWith(
                     color: Theme.of(context).colorScheme.onSurface,
                   ),
               controller: walletAddressController,
-              decoration: const InputDecoration(
-                labelText: 'Stellar Payout Address',
-              )),
-          subtitle: const Text('This address will be used for payout.'),
-          trailing: IconButton(
-              onPressed: () {
-                Clipboard.setData(
-                    ClipboardData(text: walletAddressController.text));
-                ScaffoldMessenger.of(context).clearSnackBars();
-                ScaffoldMessenger.of(context)
-                    .showSnackBar(const SnackBar(content: Text('Copied!')));
+              onChanged: (value) {
+                validateStellarAddress(value);
               },
-              icon: const Icon(Icons.copy)),
+              decoration: InputDecoration(
+                  errorText: addressError,
+                  labelText: 'Stellar Payout Address',
+                  suffixIcon: edit
+                      ? IconButton(
+                          onPressed: () {
+                            Navigator.of(context).push(MaterialPageRoute(
+                              builder: (context) => ContractsScreen(
+                                  chainType: chainType,
+                                  currentWalletAddress:

```

Comment:
after a successful update, the `widget.farm.walletAddress` is still the same as the old one, it should be updated to not show the current wallet in the contacts list